API.md

```diff
@@ -2871,6 +2871,28 @@ const handler = function (request, reply) {
 };
 ```
 
+If the handler returns a Promise then Hapi will register a `catch` handler on the promise object to catch unhandled promise rejections. The handler will `reply` with the error:
+
+```js
+const handler = function (request, reply) {
+
+  const badPromise = () => {
+
+    new Promise(() => {
+
+      setTimeout(() => throw new Error(), 1000);
```

Comment:
I guess it's to show that errors thrown from asynchronous functions inside Promises are caught and handled correctly.  By extension, synchronous errors will be caught and handled correctly too.  (?)  Happy to be corrected.